### ISSUE TITLE
Did you Save the old 6800 Source Code for your 6800 Evaluation Kit?

### ISSUE THREAD

**Issue by qNI794C28y0lrPwFsBMb7** [nickgammon/G-Pascal#1]
Mr Gammon:

     I read the article on G-Pascal on SuperCoders - did you keep your original 6800 source code for your 6800 Evaluation Kit?

     Reason I ask is I would like to recreate a Pascal compiler for the 6800 that might be ported to a SWTPCo 6800.

Regards,

Nelson

**Comment by nickgammon** [comment]
No, I didn't. Not really. It was on magnetic tape which I still possess, but I tried to read it a couple of years ago and it was mainly crap. And in those days I didn't have a way of printing it.

I suggest you look at the code which I ported to the 6502, which I see you have found. :)

The processors are different, but not that much different.  As I recall, the 6800 had an A and B register, and a 2-byte X register for indexing. The 6502 works somewhat differently, but if I could port it from 6800 to 6502, then you can port it back. ;)


Also see this link for more documentation, syntax diagrams and so on: https://gammon.com.au/G-Pascal/

Regards,

Nick

**Comment by qNI794C28y0lrPwFsBMb7** [comment]
Nick:

     Ah, well, I could hope :-)

     I will have to dust off the 6502 programming books I have to understand the X and Y index registers a bit better.

     Thanks for the response -

Regards,


Nelson
-----Original Message-----
